File: src/main/java/siosio/doma/livetemplate/DaoLiveTemplatesProvider.java
package siosio.doma.livetemplate;

import com.intellij.codeInsight.template.impl.DefaultLiveTemplatesProvider;
import org.jetbrains.annotations.Nullable;

/**
 * Live Templatesを提供する。
 */
public class DaoLiveTemplatesProvider implements DefaultLiveTemplatesProvider {

    @Override
    public String[] getDefaultLiveTemplateFiles() {
        return new String[]{"doma.xml"};
    }

    @Nullable
    @Override
    public String[] getHiddenLiveTemplateFiles() {
        return new String[0];
    }
}

File: src/main/java/siosio/doma/livetemplate/DaoLiveTemplatesProvider.kt
package siosio.doma.livetemplate

import com.intellij.codeInsight.template.impl.*

/**
 * DAOメソッド用のテンプレートを提供する。
 */
class DaoLiveTemplatesProvider : DefaultLiveTemplatesProvider {

  override fun getDefaultLiveTemplateFiles(): Array<String> = arrayOf("doma")

  override fun getHiddenLiveTemplateFiles(): Array<String>? = emptyArray()
}
